src/main/java/games/strategy/triplea/ui/menubar/LobbyMenu.java

```diff
@@ -298,38 +303,38 @@ private long requestTimespanSupplication() {
             JOptionPane.OK_CANCEL_OPTION, JOptionPane.QUESTION_MESSAGE, null, timeUnits.toArray(),
             timeUnits.toArray()[3]);
     if (result < 0) {
-      return -1;
+      return Duration.ofMillis(-1);
     }
     final String selectedTimeUnit = (String) timeUnits.toArray()[result];
     if (selectedTimeUnit.equals("Forever")) {
-      return Long.MAX_VALUE;
+      return null;
     }
-    final String stringr = JOptionPane.showInputDialog(
-        lobbyFrame,
-        "Now please enter the length of time: (In " + selectedTimeUnit + "s) ",
-        1);
+    final String stringr = JOptionPane.showInputDialog(lobbyFrame,
+        "Now please enter the length of time: (In " + selectedTimeUnit + "s) ", 1);
     if (stringr == null) {
-      return -1;
+      return Duration.ofMillis(-1);
     }
     final long result2 = Long.parseLong(stringr);
     if (result2 < 0) {
-      return -1;
+      return Duration.ofMillis(-1);
     }
-    long ticks = 0;
-    if (selectedTimeUnit.equals("Minute")) {
-      ticks = result2 * 1000 * 60;
-    } else if (selectedTimeUnit.equals("Hour")) {
-      ticks = result2 * 1000 * 60 * 60;
-    } else if (selectedTimeUnit.equals("Day")) {
-      ticks = result2 * 1000 * 60 * 60 * 24;
-    } else if (selectedTimeUnit.equals("Week")) {
-      ticks = result2 * 1000 * 60 * 60 * 24 * 7;
-    } else if (selectedTimeUnit.equals("Month")) {
-      ticks = result2 * 1000 * 60 * 60 * 24 * 30;
-    } else if (selectedTimeUnit.equals("Year")) {
-      ticks = result2 * 1000 * 60 * 60 * 24 * 365;
+
+    switch (selectedTimeUnit) {
+      case "Minute":
+        return Duration.ofMinutes(result2);
+      case "Hour":
+        return Duration.ofHours(result2);
+      case "Day":
+        return Duration.ofDays(result2);
+      case "Week":
+        return Period.ofWeeks((int) result2);
+      case "Month":
+        return Period.ofMonths((int) result2);
+      case "Year":
+        return Period.ofYears((int) result2);
+      default:
+        throw new IllegalArgumentException("Invalid time unit: " + selectedTimeUnit);
```

Comment:
Not sure `IllegalArgumentException` is the most appropriate exception type here.  I used it over in `LobbyFrame` because the value I was switching on was an input parameter to the method.  Here it's all internal to this method.  `AssertionError` might be more appropriate because you are defining all the possible values locally to this method in the `timeUnits` collection.